snow/engine/snowman/transitive_test.go

```diff
@@ -3009,157 +2997,136 @@ func TestEngineBubbleVotesThroughInvalidChain(t *testing.T) {
-						return blk1, nil
-					default:
-						t.Fatalf("Unknown block bytes")
-						return nil, nil
-					}
-				}
-
-				// The VM should only be able to retrieve [gBlk] from storage
-				vm.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
-					switch blkID {
-					case gBlk.ID():
-						return gBlk, nil
-					default:
-						return nil, errUnknownBlock
-					}
-				}
-
-				pullQuerySent := new(bool)
-				pullQueryReqID := new(uint32)
-				pullQueriedVdrs := set.Set[ids.NodeID]{}
-				sender.SendPullQueryF = func(_ context.Context, inVdrs set.Set[ids.NodeID], requestID uint32, blkID ids.ID) {
-					switch {
-					case *pullQuerySent:
-						t.Fatalf("Asked multiple times")
-					case blkID != blk1.ID():
-						t.Fatalf("Expected engine to request blk1")
-					}
-					pullQueriedVdrs.Union(inVdrs)
-					*pullQuerySent = true
-					*pullQueryReqID = requestID
-				}
-
-				pushQuerySent := new(bool)
-				pushQueryReqID := new(uint32)
-				pushQueriedVdrs := set.Set[ids.NodeID]{}
-				sender.SendPushQueryF = func(_ context.Context, inVdrs set.Set[ids.NodeID], requestID uint32, blkBytes []byte) {
-					switch {
-					case *pushQuerySent:
-						t.Fatal("Asked multiple times")
-					case !bytes.Equal(blkBytes, blk1.Bytes()):
-						t.Fatal("got unexpected block bytes instead of blk1")
-					}
-					*pushQuerySent = true
-					*pushQueryReqID = requestID
-					pushQueriedVdrs.Union(inVdrs)
-				}
-
-				// Give the engine blk1. It should insert it into consensus and send a mixed query
-				// consisting of 12 push queries and 8 pull queries.
-				if err := te.Put(context.Background(), te.Validators.List()[0].NodeID, constants.GossipMsgRequestID, blk1.Bytes()); err != nil {
-					t.Fatal(err)
-				}
-
-				switch {
-				case !*pullQuerySent:
-					t.Fatal("expected us to send pull queries")
-				case !*pushQuerySent:
-					t.Fatal("expected us to send push queries")
-				case *pushQueryReqID != *pullQueryReqID:
-					t.Fatalf("expected equal push query (%v) and pull query (%v) req IDs", *pushQueryReqID, *pullQueryReqID)
-				case pushQueriedVdrs.Len()+pullQueriedVdrs.Len() != te.Config.Params.K:
-					t.Fatalf("expected num push queried (%d) + num pull queried (%d) to be %d", pushQueriedVdrs.Len(), pullQueriedVdrs.Len(), te.Config.Params.K)
-				case !tt.isVdr && pushQueriedVdrs.Len() != te.Params.MixedQueryNumPushNonVdr:
-					t.Fatalf("expected num push queried (%d) to be %d", pushQueriedVdrs.Len(), te.Params.MixedQueryNumPushNonVdr)
-				case tt.isVdr && pushQueriedVdrs.Len() != te.Params.MixedQueryNumPushVdr:
-					t.Fatalf("expected num push queried (%d) to be %d", pushQueriedVdrs.Len(), te.Params.MixedQueryNumPushVdr)
-				}
-
-				pullQueriedVdrs.Union(pushQueriedVdrs) // Now this holds all queried validators (push and pull)
-				for vdr := range pullQueriedVdrs {
-					if !vdrs.Contains(vdr) {
-						t.Fatalf("got unexpected vdr %v", vdr)
-					}
-				}
-			})
-	}
-}
+// func TestSendQuery(t *testing.T) {
```

Comment:
Once this test is fixed - this PR LGTM.

Reply:
Actually I looked around the repo and I think `TestEngineBuildBlock` test that we currently have is what we want instead of this test. I added a check to just make sure we're not sending pulls and only pushes and I think we can remove this test (since if we modified all test we would want a mixed query of all pushes and no pulls which is basically the aforementioned test).